dwave_networkx/algorithms/tsp.py

```diff
@@ -185,10 +195,10 @@ def traveling_salesperson_qubo(G, lagrange=None, weight='weight'):
             nextpos = (pos + 1) % N
 
             # going from u -> v
-            Q[((u, pos), (v, nextpos))] += G[u][v][weight]
+            Q[((u, pos), (v, nextpos))] += G[u][v].get(weight, missing_edge_weight)
 
             # going from v -> u
-            Q[((v, pos), (u, nextpos))] += G[u][v][weight]
+            Q[((v, pos), (u, nextpos))] += G[u][v].get(weight, missing_edge_weight)
```

Comment:
I think you want `G[v][u]` in this case, to allow different weights in different direction